ui/imports/shared/popups/keycard/helpers/KeyPairUnknownItem.qml

```diff
@@ -0,0 +1,170 @@
+                anchors.horizontalCenter: parent.horizontalCenter
+                width: parent.width - 2 * Style.current.padding
+                height: Style.current.xlPadding * 2
+                color: Theme.palette.statusModal.backgroundColor
+                radius: Style.current.halfPadding
+
+                RowLayout {
+                    anchors.verticalCenter: parent.verticalCenter
+                    anchors.left: parent.left
+                    anchors.right: parent.right
+                    anchors.leftMargin: Style.current.padding
+                    anchors.rightMargin: Style.current.padding
+
+                    ColumnLayout {
+                        Row {
+                            spacing: 0
+                            padding: 0
+                            StatusBaseText {
+                                id: address
+                                text: StatusQUtils.Utils.elideText(model.address, 6, 4)
+                                wrapMode: Text.WordWrap
+                                font.pixelSize: Constants.keycard.general.fontSize2
+                                color: Theme.palette.directColor1
+                            }
+
+                            StatusFlatRoundButton {
+                                height: 20
+                                width: 20
+                                icon.name: "external"
+                                icon.width: 16
+                                icon.height: 16
+                                onClicked: {
+                                    Qt.openUrlExternally("https://etherscan.io/address/%1".arg(model.address))
+                                }
+                            }
+                        }
+
+                        StatusBaseText {
+                            text: model.path
+                            wrapMode: Text.WordWrap
+                            font.pixelSize: Constants.keycard.general.fontSize2
+                            color: Theme.palette.baseColor1
+                        }
+                    }
+
+                    Item {
+                        Layout.fillWidth: true
+                        Layout.preferredHeight: parent.height
+                    }
+
+                    StatusBaseText {
+                        Layout.alignment: Qt.AlignVCenter
+                        text: {
+                            if (Global.appMain) {
+                                return "%1%2".arg(SharedStore.RootStore.currencyStore.currentCurrencySymbol)
+                                .arg(Utils.toLocaleString(model.balance.toFixed(2), appSettings.locale, {"model.currency": true}))
+                            }
+                            // without language/model refactor no way to read currency symbol or `appSettings.locale` before user logs in
+                            return "$%1".arg(Utils.toLocaleString(model.balance.toFixed(2), localAppSettings.language, {"model.currency": true}))
```

Comment:
Yeah... I'll need to centralize/improve the currency/number formatting functions; these mistakes just keep on repeating all over our codebase